File: frontend/src/app/task.service.ts
import { Injectable } from '@angular/core';
import { WebRequestService } from './web-request.service';

@Injectable({
  providedIn: 'root'
})
export class TaskService {

  constructor(private webService:WebRequestService) { }

  getAllTasks(){
    return this.webService.get("tasks/all");
  }
  getAllCompletedTasks(){
    return this.webService.get("tasks/completed");
  }
  getAllPendingTasks(){
    return this.webService.get("tasks/pending");
  }
  getTaskWithID(taskID:number){
    let forgeString:string="tasks/"+taskID;
    return this.webService.get(forgeString);
  }
  getProductsForTask(taskID:number){
    let forgeString:string="tasks/"+taskID+"/products";
    return this.webService.get(forgeString);
  }
  createTask(description:string,taskAuthorID:number){
    return this.webService.post('tasks/createTask',{taskAuthorID,description});
  } 
  createTaskProductItem(productID:number,amount:number,taskID:number){
    let forgeString:string="tasks/"+taskID+"/createList";
    return this.webService.post(forgeString,{productID,amount});
  } 
  acceptTask(taskID:number,employeeID:number){
    let forgeString:string="tasks/"+taskID+"/accept";
    return this.webService.put(forgeString,{employeeID});
  }
  completeTask(taskID:number,taskFinisherID:number){
    let forgeString:string="tasks/"+taskID+"/complete";
    return this.webService.put(forgeString,{taskFinisherID});
  }
}

File: src/main/java/com/warehouse/services/TaskService.java
package com.warehouse.services;

import com.warehouse.model.ProductsForTask;
import com.warehouse.model.Task;

import java.util.List;
import java.util.Map;

public interface TaskService {
    List<Task> getAllTasks();
    Task getTaskWithId(int taskID);
    List<Task> getAllCompletedTasks();
    List<Task> getAllNotCompletedTasks();
    int createTask(Map<String,Object> body);
    int createTask(Task task);
    boolean createProductListItemForTask(Map<String,Object> body, int taskID);
    Task completeTask(Map<String,Object> body,int taskD);

    List<ProductsForTask> getProductsForTaskWithID(int taskID);

    Task acceptTask(int taskID, Map<String,Object> body);
}
